api/client/admin/storagejobs.go

```diff
@@ -4,96 +4,98 @@ import (
 	"context"
 
 	adminPb "github.com/textileio/powergate/api/gen/powergate/admin/v1"
+	userPb "github.com/textileio/powergate/api/gen/powergate/user/v1"
 )
 
 // StorageJobs provides access to Powergate jobs admin APIs.
 type StorageJobs struct {
 	client adminPb.AdminServiceClient
 }
 
-type storageJobsConfig struct {
-	UserID string
-	Cids   []string
-}
+// ListSelect specifies which StorageJobs to list.
+type ListSelect int32
 
-// StorageJobsOption configures a storageJobsConfig.
-type StorageJobsOption = func(*storageJobsConfig)
+const (
+	// All lists all StorageJobs and is the default.
+	All ListSelect = iota
+	// Queued lists queued StorageJobs.
+	Queued
+	// Executing lists executing StorageJobs.
+	Executing
+	// Final lists final StorageJobs.
+	Final
+)
 
-// WithUserID filters the results to the specified user.
-func WithUserID(userID string) StorageJobsOption {
-	return func(conf *storageJobsConfig) {
-		conf.UserID = userID
-	}
+// ListConfig controls the behavior for listing StorageJobs.
+type ListConfig struct {
+	// UserIDFilter filters StorageJobs list to the specified user ID. Defaults to no filter.
+	UserIDFilter string
+	// CidFilter filters StorageJobs list to the specified cid. Defaults to no filter.
+	CidFilter string
```

Comment:
It's currently just one single cid. We previously had the ability to filter by many cids, but it seemed like something most people would probably never use, so I made it a single cid for now since that fits well with go-datastore prefix querying using those new indexes. Maybe this is too simple? Open to discussion here.

Reply:
Ooops, yea the comment is wrong, will correct.

Reply:
No never mind, it is correct.

Reply:
Oh, I see now... misread/read_too_fast. :)